### PR TITLE
Hotfix: import api lambda domain name

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

domain names for the api lambdas already exist outside terraform, need to import them.

This is a bit hacky, as originally in this PR I added an import to `terragrunt_plan_staging.yml` . But the import added `aws_api_gateway_domain_name.api_lambda` to the staging environment, so now the plan doesn't need it.

In the staging deploy I do a remove then import to recreate in case the environment variable is changed.

For now I've added the import to the production plan.

See
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_domain_name#import
https://www.terraform.io/docs/cli/commands/state/rm.html#example-remove-all-instances-of-a-resource

# Test instructions | Instructions pour tester la modification

# Help requested | Aide requise

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: .github/workflows/merge_to_main_production.yml [hunk 1]
```diff
@@ -113,4 +113,6 @@ jobs:
       - name: Apply aws/lambda-api
         run: |
           cd env/production/lambda-api
+          ../../../bin/terragrunt state rm 'aws_api_gateway_domain_name.api_lambda'
+          ../../../bin/terragrunt import aws_api_gateway_domain_name.api_lambda ${{ secrets.PRODUCTION_API_LAMBDA_DOMAIN_NAME }}
```
Comment (jimleroyer) [inline .github/workflows/merge_to_main_production.yml:117]: Are these import meant to be executed once only?

File: .github/workflows/terragrunt_plan_production.yml [hunk 2]
```diff
@@ -155,6 +155,13 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
           terragrunt: "true"
 
+      - name: import aws_api_gateway_domain_name for api lambda
+        if: ${{ steps.filter.outputs.lambda-api == 'true' || steps.filter.outputs.common == 'true' }}
+        run: |
+          cd env/staging/lambda-api
+          ../../../bin/terragrunt import aws_api_gateway_domain_name.api_lambda ${{ secrets.PRODUCTION_API_LAMBDA_DOMAIN_NAME }}
+
```
Comment (sastels) [inline .github/workflows/terragrunt_plan_production.yml:163]: This block should be removed after it runs the first time.

Comment (patheard) [reply .github/workflows/terragrunt_plan_production.yml:163]: Another option would be to just do the import locally so you don't have to have these in the workflow files.

File: .github/workflows/merge_to_main_production.yml (hunk 1 above)
Comment (sastels) [reply .github/workflows/merge_to_main_production.yml:117]: no, my thinking is that if `PRODUCTION_API_LAMBDA_DOMAIN_NAME` changes and these lines are not here, then the app will not be connected to the new url.

The import in `terragrunt_plan_production.yml` should only be run once. (I've just added a comment 😅 )

File: .github/workflows/terragrunt_plan_production.yml (hunk 2 above)
Comment (jimleroyer) [reply .github/workflows/terragrunt_plan_production.yml:163]: I know Antoine did that for manual import, but I never did it myself. That would be an interesting exercise.

**Review by jimleroyer — APPROVED** [review]

**Comment by sastels** [comment]
As discussed offline with @patheard, this PR is not needed, and a more traditional approach is to do the import once manually.

 Note that I’ve reran the merge to staging action and it succeeded.

**Pull request closed without merging**